### ISSUE TITLE
V1.3.5 on MacOS 15.1.1 seems to have stopped working since upgrade to 15.1.1

### ISSUE THREAD

**Issue by deepcoder** [iina/iina#5280]
### IINA Version

1.3.5

### Release Channel

Release

### macOS Version

15.1.1

### Steps to Reproduce

Just get a bouncing icon in MacOS doc. Tried the nightly release and MacOS will not let these run. Tried deleting app in Applications folder and reinstalling. Same behavior. Deleted and tried installing via brew, same behavior.

mpv version 0.39.0 works fine want running it from command line.

### Expected Behavior

video window

### Actual Behavior

bouncing icon in MacOS dock have to force quit

### How often can you reproduce the issue

every time

### Required Actions

- [X] I tried my best to search from the existing issues (including closed issues) and didn't find duplicated issues.

### Optional Actions

- [ ] I tested with the Nightly build and confirmed the issue is not fixed.
- [X] I tested with the latest version of mpv and confirmed mpv doesn't have this issue.

**Comment by svobs** [comment]
I upgraded this morning and see no issue starting 1.3.5 on my M1 Pro. Are you perhaps using an Intel-based Mac?

**Comment by low-batt** [comment]
I upgraded my Intel Mac to check on this issue. From the IINA log file:
```text
23:40:35.633 [iina][d] Running under macOS Version 15.1.1 (Build 24B91)
23:40:35.633 [iina][d] On a MacBookPro16,1 with an Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz processor
```

IINA 1.3.5 started and played a video without any problems for me. I did get some prompts from macOS to grant permissions.

When the IINA's icon is bouncing make sure there isn't an IINA window, such as the welcome window, hidden behind other windows on the desktop.

**Comment by deepcoder** [comment]
Thank you very much for the replies and ideas, as well as the fine work on this useful project.

There is no problem with with IINA on MacOS 15.1.1 on M2 Mac hardware. As was pointed out in above replies, there was a small dialog box indicating IINA was not able to connect to a NAS device hidden behind another window. I has a Synology NAS power brick fail during the same window between uses of IINA and the MacOS upgrade.

I cleared the watch history in IINA and the dialog no longer appears.

Thank you again for your help and work! Closing this issue.

**Comment by low-batt** [comment]
Thanks so much for reporting what was happening and closing the issue.

That IINA is triggering the mount request is a defect. The primary issue for that is #4973. A workaround can be found in [this comment](https://github.com/iina/iina/issues/4973#issuecomment-2181352809).

Briefly, in Sonoma Apple changed the macOS daemon that manages the list of recent documents seen in `Open Recent` under the `File` menu and in the Welcome window. If the code signature of the IINA executable does not match the signature of the IINA executable that last accessed the list the daemon clears the list. Only the release and beta versions of IINA are signed with the official certificate. Developers running the Debug version of IINA or users testing an IINA Nightly build will have this list cleared. I did not find any mention of this change in macOS release notes.

A workaround was added to save and restore the list when this happens. Unfortunately it was not noticed that if the list contains references to files on external devices macOS would either ask for the device to be mounted or wait on a long timeout for the device to response when the workaround restored the list. This workaround has been disabled by default in the development version of IINA. We are getting close to a beta release with this fix.

The workaround kicks in when it detects the list of recent documents is empty and the saved copy is not. The expectation is that this should only happen when someone runs something other than the release version of IINA. And yet we have cases reported where apparently the list was cleared where the users did not run a non-release version of IINA. It is troubling that as of yet we can't explain this aspect of the problem.
